ui/apps/dashboard/src/app/(organization-active)/(dashboard)/settings/integrations/page.tsx

```diff
@@ -1,10 +1,62 @@
 import { redirect } from 'next/navigation';
+import { auth } from '@clerk/nextjs';
 
 import { getBooleanFlag } from '@/components/FeatureFlags/ServerFeatureFlag';
+import graphqlAPI from '@/queries/graphqlAPI';
+import { getProductionEnvironment } from '@/queries/server-only/getEnvironment';
 import IntegrationsList from './integrations';
+import type { VercelIntegration } from './vercel/VercelIntegration';
+import mergeVercelProjectData from './vercel/mergeVercelProjectData';
+import { GetSavedVercelProjectsDocument } from './vercel/queries';
+
+export const notEnabledVercelIntegration: VercelIntegration = {
+  id: 'not-enabled',
+  name: 'Vercel',
+  slug: 'vercel',
+  projects: [],
+  enabled: false,
+};
+
+export const vercelIntegration = async () => {
+  const { getToken } = auth();
+  const sessionToken = await getToken();
+  const { id: environmentID } = await getProductionEnvironment();
+  const {
+    environment: { savedVercelProjects: savedProjects = [] },
+  } = await graphqlAPI.request(GetSavedVercelProjectsDocument, {
+    environmentID,
+  });
+  const url = new URL('/v1/integrations/vercel/projects', process.env.NEXT_PUBLIC_API_URL);
+  url.searchParams.set('workspaceID', environmentID);
+
+  const restResponse = await fetch(url, {
+    method: 'GET',
+    headers: {
+      Authorization: `Bearer ${sessionToken}`,
+    },
+  });
+
+  const data = await restResponse.json();
```

Comment:
`data` has an implicit type of `any` here. Can you use a zod schema to ensure that our static types match runtime types?

Reply:
Yup, see the latest. I used a zod schema, inferred the response type from that and validated that the response shape matched the schema at runtime.